### PR TITLE
Create regex-2019.12.20.recipe

### PR DESCRIPTION
**Pull request by ubports-on-cedric**
Regex compiles and works ok on x86_64

### CODE REVIEW

**Comment by extrowerk** [comment]
Why your recipe called as "2019.20.12"?
Currently your recipe is broken, as it tries to download a file called "regex-2019.20.12.tar.gz", while it is called as "regex-2019.12.20.tar.gz" at the server.
Please fix the name of the recipe.

**Comment by extrowerk** [comment]
After fixing the filename the recipe builds fine, import tested with Python 3.6 and 3.7, works.

**Comment by Begasus** [comment]
> Why your recipe called as "2019.20.12"?
> Currently your recipe is broken, as it tries to download a file called "regex-2019.20.12.tar.gz", while it is called as "regex-2019.12.20.tar.gz" at the server.
> Please fix the name of the recipe.

That was my bad, a bit dislectic here I think :) But building it would have revealed that.

**Comment by Begasus** [comment]
Please add secondary architecture here

File: dev-python/regex/regex-2019.20.12.recipe [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+HOMEPAGE="https://pypi.org/project/regex/"
+COPYRIGHT="2013-2019 Matthew Barnett"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://pypi.org/packages/source/r/regex/regex-$portVersion.tar.gz"
+CHECKSUM_SHA256="106e25a841921d8259dcef2a42786caae35bc750fb996f830065b3dfaa67b77e"
+SOURCE_DIR="regex-2019.12.20"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+
```
Comment (Begasus) [inline dev-python/regex/regex-2019.20.12.recipe]: Missing something here to be able to build it for secondary architectures

File: dev-python/regex/regex-2019.20.12.recipe [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+DESCRIPTION="Regular expression module, to replace re"
+HOMEPAGE="https://pypi.org/project/regex/"
+COPYRIGHT="2013-2019 Matthew Barnett"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://pypi.org/packages/source/r/regex/regex-$portVersion.tar.gz"
+CHECKSUM_SHA256="106e25a841921d8259dcef2a42786caae35bc750fb996f830065b3dfaa67b77e"
+SOURCE_DIR="regex-2019.12.20"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
```
Comment (Begasus) [inline dev-python/regex/regex-2019.20.12.recipe]: Missing something here to be able to build it for secondary architectures

Comment (ubports-on-cedric) [reply dev-python/regex/regex-2019.20.12.recipe]: like now?

**Comment by Begasus** [comment]
LGTM @korli @extrowerk could you check also?

File: dev-python/regex/regex-2019.12.20.recipe [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+REVISION="1"
+SOURCE_URI="https://pypi.org/packages/source/r/regex/regex-$portVersion.tar.gz"
+CHECKSUM_SHA256="106e25a841921d8259dcef2a42786caae35bc750fb996f830065b3dfaa67b77e"
+SOURCE_DIR="regex-2019.12.20"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	$portName = $portVersion
```
Comment (Begasus) [inline dev-python/regex/regex-2019.12.20.recipe]: My bad, you  are missing some $secondaryArchSuffix in the recipe (4 places)

File: dev-python/regex/regex-2019.12.20.recipe [hunk 4]
```diff
@@ -0,0 +1,64 @@
+SUMMARY="All things regular expression: implemented in python"
+DESCRIPTION="Regular expression module, to replace re"
+HOMEPAGE="https://pypi.org/project/regex/"
+COPYRIGHT="2013-2019 Matthew Barnett"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://pypi.org/packages/source/r/regex/regex-$portVersion.tar.gz"
+CHECKSUM_SHA256="106e25a841921d8259dcef2a42786caae35bc750fb996f830065b3dfaa67b77e"
+SOURCE_DIR="regex-2019.12.20"
```
Comment (korli) [inline dev-python/regex/regex-2019.12.20.recipe]: Is SOURCE_DIR really needed?

File: dev-python/regex/regex-2019.12.20.recipe (hunk 3 above)
Comment (ubports-on-cedric) [reply dev-python/regex/regex-2019.12.20.recipe]: like now?

File: dev-python/regex/regex-2019.12.20.recipe (hunk 4 above)
Comment (ubports-on-cedric) [reply dev-python/regex/regex-2019.12.20.recipe]: Yes, it failes without it at least onx86_64.

File: dev-python/regex/regex-2019.12.20.recipe (hunk 3 above)
Comment (Begasus) [reply dev-python/regex/regex-2019.12.20.recipe]: $portName should be $secondaryArchSuffix too

File: dev-python/regex/regex-2019.12.20.recipe (hunk 4 above)
Comment (korli) [reply dev-python/regex/regex-2019.12.20.recipe]: The current value is the default value, si it shouldn't be needed.

**Comment by korli** [comment]
@Begasus ok to merge?

**Comment by Begasus** [comment]
@korli just doing a final check here for _x86

**Comment by Begasus** [comment]
LGTM :)

File: dev-python/regex/regex-2019.12.20.recipe [hunk 5]
```diff
@@ -0,0 +1,63 @@
+SUMMARY="All things regular expression: implemented in python"
+DESCRIPTION="Regular expression module, to replace re"
```
Comment (Begasus) [inline dev-python/regex/regex-2019.12.20.recipe]: Maybe a better description?
This regex implementation is backwards-compatible with the standard ‘re’ module, but offers additional functionality.

Comment (Begasus) [reply dev-python/regex/regex-2019.12.20.recipe]: Just caught my eye when opening in HD :)

Comment (Begasus) [reply dev-python/regex/regex-2019.12.20.recipe]: @korli I can't push to remote branches here (at least never did), could you push some change to the description, other then that the PR is OK for me

**Pull request merged**